### PR TITLE
Add --show-report option to charm-build

### PR DESCRIPTION
**Pull request by johnsca**
This option enables a post-build report detailing what files have changed, based on the signatures in the existing `.build.manifest`

### CODE REVIEW

**Comment by chuckbutler** [comment]
YES!

**Comment by johnsca** [comment]
Making some refactors to use existing delta_signatures logic as suggested in IRC.

File: charmtools/build/__init__.py [hunk 1]
```diff
@@ -588,6 +623,8 @@ def main(args=None):
                         default="http://interfaces.juju.solutions")
     parser.add_argument('-n', '--name',
                         help="Build a charm of 'name' from 'charm'")
+    parser.add_argument('-r', '--show-report', action="store_true",
```
Comment (marcoceppi) [inline charmtools/build/__init__.py]: NIT: `--report` or `--with-report`?

File: charmtools/build/__init__.py [hunk 2]
```diff
@@ -518,6 +520,39 @@ def clean_removed(self, signatures):
...
             filepath.remove()
 
+    def write_report(self, new_sigs):
+        """
+        Log messages indicating what changed with this (re)build.
+        """
+        old_sigs = self.old_manifest['signatures']
+        if not old_sigs:
+            log.info('')
+            log.info('---------------------------------------')
```
Comment (marcoceppi) [inline charmtools/build/__init__.py]: Would we ever want report data to be machine readable?

**Comment by johnsca** [comment]
Made some changes based on review.

@marcoceppi The report format is basically machine-readable but is still mixed in with the log output.  I suppose we might want to dump it to a file at some point with less header info, but at that point it could maybe just  be added to the `.build.manifest` file in a "changes" section or something.

**Comment by marcoceppi** [comment]
LGTM :BOAT:

**Comment by bcsaller** [comment]
LGTM, thanks

**Pull request merged**